data_store/table_extraction/process.py

```diff
@@ -41,34 +42,19 @@ class TableProcessor:
     drop_empty_tables: bool
     dropdown_placeholder: str
 
-    def __init__(
-        self,
-        num_header_rows: int = 1,
-        ignored_non_header_rows: list[int] | None = None,
-        merged_header_rows: list[int] | None = None,
-        col_names_to_drop: list[str] | None = None,
-        drop_empty_rows: bool = False,
-        drop_empty_tables: bool = False,
-        dropdown_placeholder: str = "< Select >",
-    ):
-        if ignored_non_header_rows is None:
-            ignored_non_header_rows = []
-        if merged_header_rows is None:
-            merged_header_rows = []
-        if col_names_to_drop is None:
-            col_names_to_drop = []
-        if any(header_idx >= num_header_rows for header_idx in merged_header_rows):
+    def __init__(self, process_config: ProcessConfig):
+        if any(header_idx >= process_config.num_header_rows for header_idx in process_config.merged_header_rows):
             raise TableProcessingError(
-                f"Merged header row indexes {merged_header_rows} must be with the range of specified headers "
-                f"(0-{num_header_rows - 1})"
+                f"Merged header row indexes {process_config.merged_header_rows} must be with the range of specified "
+                f"headers (0-{process_config.num_header_rows - 1})"
             )
-        self.num_header_rows = num_header_rows
-        self.ignored_non_header_rows = ignored_non_header_rows
-        self.merged_header_rows = merged_header_rows
-        self.col_names_to_drop = col_names_to_drop
-        self.drop_empty_rows = drop_empty_rows
-        self.drop_empty_tables = drop_empty_tables
-        self.dropdown_placeholder = dropdown_placeholder
+        self.num_header_rows = process_config.num_header_rows
+        self.ignored_non_header_rows = process_config.ignored_non_header_rows
+        self.merged_header_rows = process_config.merged_header_rows
+        self.col_names_to_drop = process_config.col_names_to_drop
+        self.drop_empty_rows = process_config.drop_empty_rows
+        self.drop_empty_tables = process_config.drop_empty_tables
+        self.dropdown_placeholder = process_config.dropdown_placeholder
```

Comment:
nit: it maybe feels a bit weird to have a container for config in `ProcesConfig` but then just dupe all of the attributes - should we just store the process_config object and redirect through that?

It's probably fine for now and we could address in the future if needed.